kernel/kernel-api/src/main/java/io/delta/kernel/fs/FileStatus.java

```diff
@@ -0,0 +1,86 @@
+package io.delta.kernel.fs;
+
+import java.util.Objects;
+
+/**
+ * Class for encapsulating metadata about a file in Delta Lake table.
+ */
+public class FileStatus {
+
+    private final String path;
+    private final long size;
+    private final long modificationTime;
+    private final boolean hasDeletionVector;
```

Comment:
`hasDeletionVector` is not needed. I think we can remove it. If needed we can add it later.

Reply:
sgtm